app/controllers/stories_controller.rb

```diff
@@ -330,4 +331,58 @@ def assign_podcasts
   def assign_classified_listings
     @classified_listings = ClassifiedListing.where(published: true).select(:title, :category, :slug, :bumped_at)
   end
+
+  def set_json_ld
+    @user_json_ld = {
+      "@context": "http://schema.org",
+      "@type": "Person",
+      "mainEntityOfPage": {
+        "@type": "WebPage",
+        "@id": "https://dev.to/<%= user_url(@user) %>"
+      },
+      "url": "<%= user_url(@user) %>",
+      "sameAs": [],
+      "image": "<%= ProfileImage.new(@user).get(width: 320) %>",
+      "name": "<%= @user.name %>",
+      "email": "",
+      "jobTitle": "",
+      "description": "<%= @user.summary.presence || ['404 bio not found'].sample %>",
+      "disambiguatingDescription": [],
+      "worksFor": [
+        {
+          "@type": "Organization"
+        },
+      ],
+      "alumniOf": ""
+    }
+    user_profile_details_json_ld
+    user_same_as_json_ld
+  end
+
+  def user_profile_details_json_ld
+    @user_json_ld["disambiguatingDescription"]&.append(@user.mostly_work_with) if @user.mostly_work_with.present?
+    @user_json_ld["disambiguatingDescription"]&.append(@user.currently_hacking_on) if @user.currently_hacking_on.present?
+    @user_json_ld["disambiguatingDescription"]&.append(@user.currently_learning) if @user.currently_learning.present?
+    @user_json_ld["worksFor"]&.append(@user.employer_name) if @user.employer_name.present?
+    @user_json_ld["worksFor"]&.append(@user.employer_url) if @user.employer_url.present?
+    @user_json_ld["alumniOf"]&.append(@user.education) if @user.education.present?
+    @user_json_ld["email"]&.append(@user.email) if @user.email_public
+    @user_json_ld["jobTitle"]&.append(@user.employment_title) if @user.employment_title.present?
+    @user_json_ld["sameAs"]&.append(@user.twitter_username) if @user.twitter_username.present?
+    @user_json_ld["sameAs"]&.append(@user.github_username) if @user.github_username.present?
+  end
+
+  def user_same_as_json_ld
+    @user_json_ld["sameAs"]&.append(@user.mastodon_url) if @user.mastodon_url.present?
+    @user_json_ld["sameAs"]&.append(@user.facebook_url) if @user.facebook_url.present?
+    @user_json_ld["sameAs"]&.append(@user.linkedin_url) if @user.linkedin_url.present?
+    @user_json_ld["sameAs"]&.append(@user.behance_url) if @user.behance_url.present?
+    @user_json_ld["sameAs"]&.append(@user.stackoverflow_url) if @user.stackoverflow_url.present?
+    @user_json_ld["sameAs"]&.append(@user.dribbble_url) if @user.dribbble_url.present?
+    @user_json_ld["sameAs"]&.append(@user.medium_url) if @user.medium_url.present?
+    @user_json_ld["sameAs"]&.append(@user.gitlab_url) if @user.gitlab_url.present?
+    @user_json_ld["sameAs"]&.append(@user.instagram_url) if @user.instagram_url.present?
+    @user_json_ld["sameAs"]&.append(@user.twitch_username) if @user.twitch_username.present?
+    @user_json_ld["sameAs"]&.append(@user.website_url) if @user.website_url.present?
+  end
```

Comment:
This method exists because I ran into a Metrics/CyclomaticComplexity(RuboCop) error when I initially tried adding this code into the `set_json_ld` method. This is also why the `twitter_username` and `github_username` exist in the `user_profile_details_json_ld` and not in the `user_same_as_json_ld` method. 😄